apis/components/v1alpha1/kserve_types.go

```diff
@@ -41,6 +41,14 @@ const (
 	RawDeployment DefaultDeploymentMode = "RawDeployment"
 )
 
+// +kubebuilder:validation:Pattern=`^(Headless|Headed)$`
+type RawServiceConfig string
+
+const (
+	Headless RawServiceConfig = "Headless"
```

Comment:
since this costs and types are  part  of a flat package along with other components, maybe we should use a prefix for the name,